### PR TITLE
Uml 3023 service api implementation of callback handling

### PR DESCRIPTION
**Pull request by cooperaj**
# Purpose

This will actually handle the bulk of the authentication/authorization. It takes the info given by One Login, makes a number of requests itself back to One Logon to pull user information, validates JWT signatures and such (though the Facile OpenID library should handle the majority of the integration effort)

Fixes UML-3023

## Approach

Attempts to wrap and isolate as much of the used 3rd party library code as possible to aid testing but the nature of it is that some parts of the new code are marked to ignore code coverage.

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [ ] New event_codes have been documented on the [wiki page](https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* ~I have added welsh translation tags and updated translation files~
* ~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
* ~I have notified the Interaction Designer of any content changes so that appropriate screenshots/flow diagram changes can be made~
* ~The product team have tested these changes~

### CODE REVIEW

File: mock-integrations/secrets-manager/onelogin_mock_public_key.pem
Comment (cooperaj) [inline mock-integrations/secrets-manager/onelogin_mock_public_key.pem:1]: This key is derived from the hardcoded one that is a part of the mock-onelogin-service source code. To ease developer setup it's just going to be easier to commit this. Until the mock is changed to allow keys to be supplied at runtime.

File: service-api/app/src/App/src/Handler/OneLoginAuthenticationCallbackHandler.php [hunk 1]
```diff
@@ -28,13 +29,33 @@ public function __construct()
...
+            throw new BadRequestException('An auth session must be provided');
+        }
+
+        if (empty($requestData['auth_session']['state'])) {
+            throw new BadRequestException('The auth session must contain a state');
+        }
+
+        if (empty($requestData['auth_session']['nonce'])) {
+            throw new BadRequestException('The auth session must contain a nonce');
+        }
```
Comment (MishNajam) [inline service-api/app/src/App/src/Handler/OneLoginAuthenticationCallbackHandler.php:53]: Add redirect_uri check

File: service-api/app/src/App/src/Service/Authentication/KeyPairManager/AbstractKeyPairManager.php [hunk 2]
```diff
@@ -2,46 +2,45 @@
...
+namespace App\Service\Authentication\KeyPairManager;
 
 use Aws\SecretsManager\Exception\SecretsManagerException;
 use Aws\SecretsManager\SecretsManagerClient;
 use ParagonIE\HiddenString\HiddenString;
 use Psr\Log\LoggerInterface;
 use RuntimeException;
 
-class KeyPairManager
+class AbstractKeyPairManager
```
Comment (MishNajam) [inline service-api/app/src/App/src/Service/Authentication/KeyPairManager/AbstractKeyPairManager.php:13]: mark class as Abstract

File: service-api/app/src/App/src/Service/Authentication/Token/OutOfBandCoreIdentityVerifier.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        $this->mandatoryClaims = ['sub', 'vc'];
+
+        $this->claimCheckers = [
+            new IssuerChecker([$issuer], true),
+            new IssuedAtChecker(0, true, $clock),
+            new AudienceChecker($clientId, true),
+            new ExpirationTimeChecker(0, false, $clock),
+            new NotBeforeChecker(0, true, $clock),
+            // Add 'vot' == P2 custom checker
```
Comment (MishNajam) [inline service-api/app/src/App/src/Service/Authentication/Token/OutOfBandCoreIdentityVerifier.php:52]: This needs to be added

**Review by MishNajam — APPROVED** [review]

**Pull request merged**